include/time_series/internal/base.hxx

```diff
@@ -279,13 +279,20 @@ template <typename P, typename T>
 void TimeSeriesBase<P, T>::monitor_signal()
 {
     constexpr double SLEEP_DURATION_MS = 100;
+    std::shared_ptr<ConditionVariable<P> > local_condition_ptr = condition_ptr_;
```

Comment:
Might be worth adding a comment for future readers that the local copy is needed to avoid a segfault.

Reply:
I will add a proper comment. I could not reproduce the segfault in my computer... I hope it will be the same in bamboo.

Reply:
I just tested on my end.  With this change, I do not get the segfault anymore :)